msteams-platform/toolkit/add-single-sign-on.md

```diff
@@ -0,0 +1,346 @@
+   // for JavaScript:
+export async function showUserImage(context, ssoToken, param) {
+    await context.sendActivity("Retrieving user photo from Microsoft Graph ...");
+
+    // Init TeamsFx instance with SSO token
+    const teamsfx = new TeamsFx().setSsoToken(ssoToken);
+
+    // Update scope here. For example: Mail.Read, etc.
+    const graphClient = createMicrosoftGraphClient(teamsfx, param[0]);
+    
+    // You can add following code to get your photo:
+    // let photoUrl = "";
+    // try {
+    //   const photo = await graphClient.api("/me/photo/$value").get();
+    //   photoUrl = URL.createObjectURL(photo);
+    // } catch {
+    //   // Could not fetch photo from user's profile, return empty string as placeholder.
+    // }
+    // if (photoUrl) {
+    //   await context.sendActivity(
+    //     `You can find your photo here: ${photoUrl}`
+    //   );
+    // } else {
+    //   await context.sendActivity("Could not retrieve your photo from Microsoft Graph. Please make sure you have uploaded your photo.");
+    // }
+
+    return;
+}
+   ```
+
+---
+
+2. Register a new command
+
+   * Add the following line for new command registration using `addCommand` in `teamsSsoBot`:
+
+   ```bash
+
+   this.dialog.addCommand("ShowUserProfile", "show", showUserInfo);
+
+   ```
+
+   * Add following lines after the above line to register a new command `photo` and hook up with method `showUserImage` added above:
+
+   ```bash
+
+   // As shown here, you can add your own parameter into the `showUserImage` method
+   // You can also use regular expression for the command here
+   const scope = ["User.Read"];
+   this.dialog.addCommand("ShowUserPhoto", new RegExp("photo\s*.*"), showUserImage, scope);
+
+   ```
+
+3. Register your command in the Teams app manifest. Open `templates/appPackage/manifest.template.json`, and add following lines under `command` in `commandLists` of your bot:
+
+   ```JSON
+
+   {
+       "title": "photo",
+       "description": "Show user photo using Single Sign On feature"
+   }
+
+   ```
+</details>
+<br>
+
+## Debug your application
+
+Press F5 to debug your application. Teams Toolkit uses the Azure AD manifest file to register an Azure AD application for SSO. For Teams Toolkit local debug functionalities, see [Debug your Teams app locally](debug-local.md).
+
+## Customize Azure AD applications
+
+The [Azure AD app manifest](/azure/active-directory/develop/reference-app-manifest) allows you to customize various aspects of application registration. You can update the manifest as needed. If you need to include additional API permissions to access your desired APIs, see [API permissions to access your desired APIs](https://github.com/OfficeDev/TeamsFx/wiki/#customize-aad-manifest-template).
+To view your Azure AD application in Azure Portal, see [View Azure AD application in Azure portal](https://github.com/OfficeDev/TeamsFx/wiki/Manage-AAD-application-in-Teams-Toolkit#How-to-view-the-AAD-app-on-the-Azure-portal). 
```

Comment:
Since we have AAD Manifest doc in the doc site: https://github.com/MicrosoftDocs/msteams-docs/pull/5877; maybe we can link to the doc site directly instead of link to github?

Reply:
As PR: #5877 is not yet merged and it is not available in live site. We can link it once it gets merged.